### DISCUSSION TITLE
'#subtype' directive

### DISCUSSION THREAD

**Discussion by renlite** [odin-lang/Odin#4314] [Q&A] [answered]
What exactly is ```#suptype``` directive doing? Didn't find aynthing in the docs.
It is used here: https://github.com/odin-lang/Odin/blob/master/vendor/directx/d3d11/d3d11.odin

**Comment by Hyrtwol** [comment]
lets say you have 2 structs
```
A :: struct {
...
}
B:: struct {
    #subtype a: A,
    ...
}
DoA :: proc (a: A) ...

b := B{}
DoA(b)
```
it makes is possible to call a proc that that takes the subtype as an argument without the need for a cast.
in d3d11 it's used to operate on the IUnknown for all the d3 objects

**Reply by renlite** [reply]
Is this inheritance?

**Reply by Hyrtwol** [reply, marked as answer]
Not really, I would call it "auto-cast-to-subtype" :p inheritance would suggest you can override behaviour and we don't have methods in Odin.
